spinnaker/spinnaker.go

```diff
@@ -479,12 +479,35 @@ func (s Spinnaker) asgs(appName, account, clusterName string) (result []spinnake
 	return asgs, nil
 }
 
-// CloudProvider returns the cloud provider for a given account
-func (s Spinnaker) CloudProvider(account string) (provider string, err error) {
-	url := s.accountURL(account)
+// CloudProvider returns the cloud provider for a given account name
+func (s Spinnaker) CloudProvider(name string) (provider string, err error) {
+	account, err := s.account(name)
+	if err != nil {
+		return "", err
+	}
+
+	if account.CloudProvider == "" {
+		return "", errors.New("no cloudProvider field in response body")
+	}
+
+	return account.CloudProvider, nil
+}
+
+// account represents a spinnaker account
+type account struct {
+	CloudProvider string `json:"cloudProvider"`
+	Name          string `json:"name"`
+	Error         string `json:"error"`
+}
+
+// account returns an account by its name
+func (s Spinnaker) account(name string) (*account, error) {
+	url := s.accountsURL(true)
 	resp, err := s.client.Get(url)
+
+	// Usual HTTP checks
```

Comment:
Nit: It seems like it was not in the original code either but I believe after this error check we should also call `defer resp.Body.Close()`.

Reply:
Finally, I'm not sure what you're asking me, there's already a `defer resp.Body.Close()` in [l.513](https://github.com/Netflix/chaosmonkey/pull/56/files/1f5bf2551919a0e87f6723e491cde2895413f765#diff-7df64bbe98f3aaa527a08ee3baac15afR513). Did I miss something ?

Reply:
Ohh, yes! Its there. I must have missed it!